File: programmers/기능개발.cpp
#include <string>
#include <vector>
#include<queue>
#define MAX 100
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int v[MAX];
    for(int row=0;row<progresses.size();++row){
        int success=(MAX-progresses[row]);
        if(success%speeds[row]==0)v[row]=(success/speeds[row]);
        else v[row]=((success/speeds[row])+1);
    }
    // for(int row=0;row<progresses.size();++row){
    //     answer.push_back(v[row]);
    // }
    int value= v[0];
    int cnt=1;
    for(int i=0;i<(progresses.size()-1);++i){
        if(value<v[i+1]){
            answer.push_back(cnt);
            cnt=1;
            value=v[i+1];
        }
        else if(value>=v[i+1]){
            cnt++;
        }
    }
    answer.push_back(cnt);
    return answer;
}

File: programmers/Level 2/기능개발.js
function solution(progresses, speeds) {
    var answer = [0];
   //새로운 days라는 배열을 만들어 100-progress / speeds[index] 를 통해 각각의 progresses들이 배포가 몇일 걸리는 저장
    let days = progresses.map((progress,index)=>Math.ceil((100-progress)/speeds[index]));
    console.log(days);
    let maxDay = days[0];
    
    for(let i=0,j=0;i<days.length;++i){
        if(days[i]<=maxDay){
            answer[j]+=1;
        }else{
            maxDay=days[i];
            answer[++j]=1;
        }
    }
    return answer;
}
